File: mini_app/lib/models/booking.dart
class PassengerDetail {
  final String name;
  final String phone;

  PassengerDetail({
    required this.name,
    required this.phone,
  });

  factory PassengerDetail.fromJson(Map<String, dynamic> json) {
    return PassengerDetail(
      name: json['name'] ?? '',
      phone: json['phone'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'phone': phone,
    };
  }
}

class Booking {
  final String id;
  final String routeId;
  final String departureDate;
  final List<String> seatNumbers;
  final String userId;
  final List<PassengerDetail> passengerDetails;
  final int amount;
  final String currency;
  final String paymentReference;
  final String status;
  final String? transactionId;
  final String? paymentMethod;
  final String? ticketNumber;
  final String? qrCode;
  final DateTime createdAt;

  Booking({
    required this.id,
    required this.routeId,
    required this.departureDate,
    required this.seatNumbers,
    required this.userId,
    required this.passengerDetails,
    required this.amount,
    required this.currency,
    required this.paymentReference,
    required this.status,
    this.transactionId,
    this.paymentMethod,
    this.ticketNumber,
    this.qrCode,
    required this.createdAt,
  });

  factory Booking.fromJson(Map<String, dynamic> json) {
    return Booking(
      id: json['booking_id'] ?? '',
      routeId: json['route_id'] ?? '',
      departureDate: json['departure_date'] ?? '',
      seatNumbers: List<String>.from(json['seat_numbers'] ?? []),
      userId: json['user_id'] ?? '',
      passengerDetails: (json['passenger_details'] as List?)
          ?.map((e) => PassengerDetail.fromJson(e))
          .toList() ?? [],
      amount: json['amount']?.toInt() ?? 0,
      currency: json['currency'] ?? '',
      paymentReference: json['payment_reference'] ?? '',
      status: json['status'] ?? '',
      transactionId: json['transaction_id'],
      paymentMethod: json['payment_method'],
      ticketNumber: json['ticket_number'],
      qrCode: json['qr_code'],
      createdAt: DateTime.tryParse(json['created_at'] ?? '') ?? DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'booking_id': id,
      'route_id': routeId,
      'departure_date': departureDate,
      'seat_numbers': seatNumbers,
      'user_id': userId,
      'passenger_details': passengerDetails.map((e) => e.toJson()).toList(),
      'amount': amount,
      'currency': currency,
      'payment_reference': paymentReference,
      'status': status,
      'transaction_id': transactionId,
      'payment_method': paymentMethod,
      'ticket_number': ticketNumber,
      'qr_code': qrCode,
      'created_at': createdAt.toIso8601String(),
    };
  }

  String get formattedAmount => '$amount $currency';
}

File: backend/internal/models/booking.go
package models

import "time"

type PassengerDetail struct {
	Name  string `json:"name"`
	Phone string `json:"phone"`
}

type Booking struct {
	ID                 string             `json:"booking_id"`
	RouteID            string             `json:"route_id"`
	DepartureDate      string             `json:"departure_date"`
	SeatNumbers        []string           `json:"seat_numbers"`
	UserID             string             `json:"user_id"`
	PassengerDetails   []PassengerDetail  `json:"passenger_details"`
	Amount             int                `json:"amount"`
	Currency           string             `json:"currency"`
	PaymentReference   string             `json:"payment_reference"`
	Status             string             `json:"status"`
	TransactionID      string             `json:"transaction_id,omitempty"`
	PaymentMethod      string             `json:"payment_method,omitempty"`
	TicketNumber       string             `json:"ticket_number,omitempty"`
	QRCode             string             `json:"qr_code,omitempty"`
	CreatedAt          time.Time          `json:"created_at"`
}

type CreateBookingRequest struct {
	RouteID            string            `json:"route_id" binding:"required"`
	DepartureDate      string            `json:"departure_date" binding:"required"`
	SeatNumbers        []string          `json:"seat_numbers" binding:"required"`
	UserID             string            `json:"user_id" binding:"required"`
	PassengerDetails   []PassengerDetail `json:"passenger_details" binding:"required"`
}

type ConfirmPaymentRequest struct {
	TransactionID string `json:"transaction_id" binding:"required"`
	PaymentMethod string `json:"payment_method" binding:"required"`
	Amount        int    `json:"amount" binding:"required"`
}

type WebhookPaymentRequest struct {
	TransactionID     string    `json:"transaction_id" binding:"required"`
	BookingReference  string    `json:"booking_reference" binding:"required"`
	Status            string    `json:"status" binding:"required"`
	Amount            int       `json:"amount" binding:"required"`
	Timestamp         time.Time `json:"timestamp" binding:"required"`
}